### PR TITLE
fix: i18n strings not updating

### PR DESCRIPTION
**Pull request by thebrandonlucas**
### Describe the changes you have made in this PR

Fixed bug in #964. Added an "onLanguageChange" listener for strings that were declared outside React component. I had to abstract these into functions so they could be called again when the listener detected a change (otherwise the strings passed to the component would never change on an update).

[Here is where I found the solution](https://github.com/i18next/react-i18next/issues/835#issuecomment-486210116), which is from the creator of Reacti18next.

### Link this PR to an issue

#964 

### Type of change (Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

Visual testing to ensure strings are translating properly.

Also ran E2E and unit tests. 

2 E2E failed due to timeout issues, although I don't think these are due to my changes.

- [X] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [X] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

File: src/app/router/Welcome/Welcome.tsx [hunk 1]
```diff
@@ -71,6 +85,25 @@ function App() {
...
+  i18n.on("languageChanged", () => {
+    connectorRoutes = getConnectorRoutes();
+    routes = getRoutes(connectorRoutes);
+
+    // Update name to new language only, don't update status
+    const tempSteps: Step[] = [];
+    for (let i = 0; i < routes.length; i++) {
+      tempSteps.push({
+        id: routes[i].name,
+        status: steps[i].status,
```
Comment (thebrandonlucas) [inline src/app/router/Welcome/Welcome.tsx:98]: I'm open if anyone has ideas on how to simplify this, but I think either way we'd have to do something like this to update the strings on language change

File: src/app/router/Welcome/Welcome.tsx [hunk 2]
```diff
@@ -71,6 +85,25 @@ function App() {
...
+      });
+    }
+    setSteps(tempSteps);
+
```
Comment (thebrandonlucas) [inline src/app/router/Welcome/Welcome.tsx:102]: I'm open if anyone has ideas on how to simplify this, but I think either way we'd have to do something like this to update the strings on language change

**Comment by escapedcat** [comment]
Checked it out and tried. Looks good.

> 2 E2E failed due to timeout issues, although I don't think these are due to my changes.

They can be flakey. If they are passing in CI it's all good.

File: src/app/router/Welcome/Welcome.tsx [hunk 3]
```diff
@@ -71,6 +85,25 @@ function App() {
...
   const routesElement = useRoutes(routes);
 
+  const [languageChanged, setLanguageChanged] = useState(false);
+  i18n.on("languageChanged", () => {
+    connectorRoutes = getConnectorRoutes();
+    routes = getRoutes(connectorRoutes);
+
+    // Update name to new language only, don't update status
+    const tempSteps: Step[] = [];
+    for (let i = 0; i < routes.length; i++) {
```
[the remark this answers is not in the document]
Comment (thebrandonlucas) [reply src/app/router/Welcome/Welcome.tsx]: Good point, Done!

**Comment by bumi** [comment]
can you fix the conflicts? 
and a check from @dylancom would be good.

**Comment by thebrandonlucas** [comment]
@bumi Merge conflicts fixed!

@dylancom this should be ready to check now

**Review by escapedcat — APPROVED** [review]

**Comment by bumi** [comment]
utACK
I'll do a full detailed test in master. 
great work on all this!

**Pull request merged**